File: System Design/Key Technologies/Redis/src/log.ts
const C = {
  reset: '\x1b[0m',
  dim: '\x1b[2m',
  bold: '\x1b[1m',
  cyan: '\x1b[36m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  red: '\x1b[31m',
} as const;

export function title(text: string): void {
  console.log(`\n${C.bold}${C.cyan}━━ ${text} ${'━'.repeat(Math.max(0, 56 - text.length))}${C.reset}`);
}

/** 打印一条命令和它的返回值，对齐成 `CMD ARGS  →  result`。 */
export function cmd(command: string, result: unknown): void {
  console.log(`  ${C.green}${command.padEnd(44)}${C.reset}${C.dim}→${C.reset} ${format(result)}`);
}

export function note(text: string): void {
  console.log(`  ${C.dim}# ${text}${C.reset}`);
}

export function warn(text: string): void {
  console.log(`  ${C.yellow}! ${text}${C.reset}`);
}

export function error(text: string): void {
  console.log(`  ${C.red}✗ ${text}${C.reset}`);
}

function format(v: unknown): string {
  if (v === null) return `${C.dim}(nil)${C.reset}`;
  if (Array.isArray(v)) return `[${v.map(format).join(', ')}]`;
  if (typeof v === 'object') return JSON.stringify(v);
  if (typeof v === 'string') return `"${v}"`;
  return String(v);
}

export const sleep = (ms: number): Promise<void> => new Promise((r) => setTimeout(r, ms));

File: Full Stack Develop/Programming Language/Python/demos/log.py
"""The Python version of the minimal printing helper: prints "expression → result" aligned.

Corresponds to TypeScript's src/log.ts and Go's internal/logx/logx.go.
"""

from typing import Any

_RESET = "\033[0m"
_DIM = "\033[2m"
_BOLD = "\033[1m"
_CYAN = "\033[36m"
_GREEN = "\033[32m"
_RED = "\033[31m"


def title(text: str) -> None:
    """Print a section heading."""
    line = "━" * max(0, 40 - len(text))
    print(f"\n{_BOLD}{_CYAN}━━ {text} {line}{_RESET}")


def note(text: str) -> None:
    """Print an explanatory line (a dimmed # comment)."""
    print(f"  {_DIM}# {text}{_RESET}")


def show(expr: str, value: Any) -> None:
    """Print "expression → result", aligned as expr → value."""
    pad = max(1, 44 - len(expr))
    print(f"  {_GREEN}{expr}{' ' * pad}{_RESET}{_DIM}→{_RESET} {_format(value)}")


def error(text: str) -> None:
    """Print an error line."""
    print(f"  {_RED}✗ {text}{_RESET}")


def _format(v: Any) -> str:
    if v is None:
        return "None"
    if isinstance(v, str):
        return f'"{v}"'
    return repr(v)
